base/cvd/cuttlefish/host/commands/cvd/instances/stop.cpp

```diff
@@ -220,4 +194,37 @@ Result<void> ForcefullyStopGroup(const uid_t any_id_in_group) {
   return {};
 }
 
+Result<void> RunStopCvd(StopCvdParams params) {
+  const auto& stopper_path = params.bin_path;
+  cvd_common::Envs stop_cvd_envs;
+  stop_cvd_envs["HOME"] = params.home_dir;
+  // stop_cvd is located at $ANDROID_HOST_OUT/bin/stop_cvd
+  std::string android_host_out =
+      android::base::Dirname(android::base::Dirname(stopper_path));
+  stop_cvd_envs[kAndroidHostOut] = android_host_out;
+  stop_cvd_envs[kAndroidSoongHostOut] = android_host_out;
+  auto config_file_path = CF_EXPECT(GetCuttlefishConfigPath(params.home_dir));
+  stop_cvd_envs[kCuttlefishConfigEnvVarName] = config_file_path;
+  cvd_common::Args args;
+  std::string wait_flag =
+      fmt::format("--wait_for_launcher={}", params.wait_for_launcher_secs);
+  args.push_back(wait_flag);
```

Comment:
nit: `std::move(wait_flag)`

Reply:
It's intentionally copied, if the command fails it will be used again.